src/zarr/indexing.py

```diff
@@ -118,6 +120,16 @@ def is_integer_list(x: Any) -> TypeGuard[list[int]]:
     return isinstance(x, list) and len(x) > 0 and is_integer(x[0])
 
 
+def is_bool_list(x: Any) -> TypeGuard[list[int]]:
+    """True if x is a list of boolean.
+
+    This function assumes ie *does not check* that all elements of the list
+    have the same type. Mixed type lists will result in other errors that will
+    bubble up anyway.
```

Comment:
curious if we need to make this optimization at this point.

Reply:
probably not. I have tested performance differences of accessing with mask selection using boolean lists and I had a loss of performance only noticable from size 10^5 onwards (50% slower at 10^5 and 400% slower at 10^6)
But at this point you should probably not use numpy arrays instead of python list anyway